holidays/countries/thailand.py

```diff
@@ -379,24 +384,29 @@ def _add_with_observed(dt, holiday_name) -> None:
         # *** NOTE: only observed by financial and private sectors
 
         if year >= 1974:
-            _add_with_observed(date(year, MAY, 1), "National Labour Day")
+            _add_with_observed(
+                date(year, MAY, 1), self.tr("วันแรงงานแห่งชาติ")
+            )
 
         # !!! National Day (24 June) !!!
         # วันชาติ
         # Status: Defunct (Historical)
         # Starts in 1939 (B.E. 2482) by Plaek Phibunsongkhram
         # Replaced by Rama IX's birthday in 1960 (B.E. 2503) by Sarit Thanarat
         # TODO: Add check for 1939 if we support earlier dates.
+        national_day = self.tr("วันชาติ")
 
         if year <= 1959:
-            _add_with_observed(date(year, JUN, 24), "National Day")
+            _add_with_observed(date(year, JUN, 24), national_day)
```

Comment:
Couldn't make it as a suggestion for some reason:
`_add_with_observed(date(year, JUN, 24) if year <= 1959 else date(year, DEC, 5), national_day)`

Reply:
Added.